tfjs-layers/src/engine/base_random_layer.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright 2022 CodeSmith LLC
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ * =============================================================================
+ */
+
+import { LayerArgs, Layer } from './topology';
+import { randomGamma, randomNormal} from '@tensorflow/tfjs-core';
+import { randomStandardNormal, randomUniform } from '@tensorflow/tfjs-core';
+import { ValueError } from '../errors';
+
+export declare interface BaseRandomLayerArgs extends LayerArgs {}
+
+export type RNGTypes = {
+  [key: string]: Function;
+};
+
+export abstract class BaseRandomLayer extends Layer {
+  // A layer handle the random number creation and savemodel behavior.
+  /** @nocollapse */
+  static className = 'RandomWidth';
+  randomGenerator: Function;
+  private rngType: string;
+
+  private readonly rngTypes: RNGTypes = {
+    gamma: randomGamma,
+    normal: randomNormal,
+    standardNormal: randomStandardNormal,
+    uniform: randomUniform
+  };
```

Comment:
If this constant is not specific to this class, it can probably be moved outside of the class. If that's the case, you might also be able to base the `RNGTypes` type off of this constant.
```ts
const RNG_TYPES = {...};
type RNGTypes = typeof RNG_TYPES;
```